pkgs/applications/networking/instant-messengers/fractal/default.nix

```diff
@@ -95,8 +81,8 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Matrix group messaging app";
     homepage = "https://gitlab.gnome.org/GNOME/fractal";
-    license = licenses.gpl3;
-    maintainers = teams.gnome.members ++ (with maintainers; [ dtzWill ]);
-    platforms = platforms.unix;
+    license = licenses.gpl3Only;
```

Comment:
There is no license header in the repo. Though it's marked as GPL-3.0+ in Flatpak metadata, it's not a valid licnese file.

Reply:
https://gitlab.gnome.org/GNOME/fractal/-/blob/5/LICENSE?ref_type=tags ?

Reply:
It's only a GPL 3.0 license. The "or later" part needs to be declared in a license header. https://spdx.org/licenses/GPL-3.0-or-later.html#licenseHeader

Reply:
License headers are somewhat out of fashion these days. Project license declaration in build system should be sufficient: https://gitlab.gnome.org/GNOME/fractal/-/blob/68291d7abcd654cce3d0bffaf61803f6296cdc9b/meson.build#L4

Reply:
If the declaration in the build system is enough, then is the license file in the repo still needed? IANAL but I thought the metadata in the build system can't replace the license and the header. It's clearly required by the license:

> To do so, attach the following notices to the program. It is safest to attach them to the start of each source file to most effectively state the exclusion of warranty; and each file should have at least the "copyright" line and a pointer to where the full notice is found.

Reply:
Copy of the license text still needs to be provided with the software when redistributed but the headers (blurbs) are just informational – they need to be kept when present but are not really required to be present in the first place in my understanding. (Actually, as I understand it, the original developers would not need to provide the copy of the license, just the people redistributing the software, but developers usually include it to make redistribution easier.)

Some recent projects tend to omit the blurbs because they are super long and copying individual files into different projects is not really that common so people can just check out the root directory for license information.

If more precise tracking of copyright ownership outside of version control metadata, [REUSE specification](https://reuse.software/) can be used ([example](https://github.com/GNOME/gnome-shell-extensions/commit/b7f285d7331cbbf0d74090b2bacf729cb32b699d)).

Reply:
I thought REUSE is a good idea.  I revert the license change for now. Maybe we should have a standard about the license declaration for nixpkgs.